### PR TITLE
Runtime initialize amazon-services classes using j.u.Random

### PR DESCRIPTION
**Pull request by jerboaa**
@gsmet another one for #14904. This one fixes `integration-tests/amazon-services`. Thoughts?

### CODE REVIEW

File: extensions/amazon-services/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AmazonServicesClientsProcessor.java [hunk 1]
```diff
@@ -37,6 +38,16 @@ void globalInterceptors(BuildProducer<AmazonClientInterceptorsPathBuildItem> pro
...
     }
 
+    @BuildStep
+    void runtimeInitialize(BuildProducer<RuntimeInitializedClassBuildItem> producer) {
+        producer.produce(
+                new RuntimeInitializedClassBuildItem("software.amazon.awssdk.core.retry.backoff.FullJitterBackoffStrategy"));
+        // This class triggers initialization of FullJitterBackoffStragegy so needs to get runtime-initialized
+        // as well
+        producer.produce(
+                new RuntimeInitializedClassBuildItem("software.amazon.awssdk.services.dynamodb.DynamoDbRetryPolicy"));
```
Comment (gsmet) [inline extensions/amazon-services/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AmazonServicesClientsProcessor.java:48]: Shouldn't this one be in the `amazon-dynamodb` extension?

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks. I added a small question that needs clarification.

File: extensions/amazon-services/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AmazonServicesClientsProcessor.java (hunk 1 above)
Comment (jerboaa) [reply extensions/amazon-services/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AmazonServicesClientsProcessor.java]: Probably, I'm having a hard time figuring out where what should go in terms of extensions. I've force-pushed an update. Let me know if that's what you meant or if everything should go into `amazon-dynamodb`...

**Review by gsmet — APPROVED** [review]
Looks good!

The gist of it is that you should add the rules in the extension providing the class.

**Comment by jerboaa** [comment]
OK, thanks!

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**